app/components/taxonomy-top-list.js

```diff
@@ -18,18 +19,35 @@ import Analytics from '../mixins/analytics';
  */
 export default Ember.Component.extend(Analytics, {
     theme: Ember.inject.service(),
-    // jscs:disable disallowQuotedKeysInObjects
-    subjectTooltips: {
-        'Architecture': 'Architectural Engineering, Construction Engineering, Environmental Design, Interior Architecture, Landscape Architecture',
-        'Arts and Humanities': 'Fine Arts, History, Music, Philosophy, Religion',
-        'Business': 'Accounting, Finance and Financial Management, Human Resource Management, Marketing, Taxation',
-        'Education': 'Curriculum and Instruction, Educational Administration and Supervision, Educational Leadership, Higher Education, Liberal Studies',
-        'Engineering': 'Biomedical Engineering and Bioengineering, Chemical Engineering, Civil and Environmental Engineering, Electrical and Computer Engineering, Mechanical Engineering',
-        'Law': 'Civil Law, Criminal Law, Legislation, State and Local Government Law, Supreme Court of the United States',
-        'Life Sciences': 'Agriculture, Genetics and Genomics, Microbiology, Physiology, Zoology',
-        'Medicine and Health Sciences': 'Anatomy, Diseases, Medical Sciences, Public Health, Veterinary Medicine',
-        'Physical Sciences and Mathematics': 'Chemistry, Computer Sciences, Earth Sciences, Physics, Statistics and Probability',
-        'Social and Behavioral Sciences': 'Anthropology, Economics, Political Science, Psychology, Sociology',
+    getSubjectChildren(parents, provider) {
+        return provider.query('taxonomies', {
+                    filter: {
+                        parents
+                    },
+                    page: {
+                        size: 4  //number of subjects to show-up in each tooltip.
+                    }
+                })
+            .then(results =>
+                results.map(result => result.get('text'))
+            )
+            .then(results =>
+                results.toArray().sort().join(', ')
+            );
+    },
+    getAllChildren(sortedList, i, provider) {
+        if (i >= 0) {
+            const subjectId = sortedList.objectAt(i).get('id');
+            const subjectChildren = this.getSubjectChildren(subjectId, provider);
+            subjectChildren.then(result => {
+                Ember.$('#' + subjectId).attr('data-hint', result); Ember.$('#' + subjectId).attr('aria-label', result);
```

Comment:
Don't do DOM manipulation. Set a property on the component and let Handlebars do the rest.